core/domain/question_domain.py

```diff
@@ -354,6 +368,14 @@ def update_language_code(self, language_code):
         """
         self.language_code = language_code
 
+    def update_linked_skill_ids(self, linked_skill_ids):
+        """Updates the linked skill ids of the question.
+
+        Args:
+            linked_skill_ids: list(str). The skill ids linked to the question.
+        """
+        self.linked_skill_ids = list(set(linked_skill_ids))
```

Comment:
Perhaps somewhere prominent -- maybe in the docstring for the class -- you might want to issue a warning to devs to not update this field manually by setting self.linked_skill_ids directly.

Could we also add a uniqueness check to the validate() method for this domain class?

Also we should add a check to validate() that the list is non-empty?

Reply:
Done